core/src/main/java/com/graphhopper/storage/TurnCostExtension.java

```diff
@@ -109,16 +111,45 @@ public boolean loadExisting() {
     /**
      * This method adds a new entry which is a turn restriction or cost information via the
      * turnFlags.
+     *
+     * It appends a new entry and does not overwrite existing entries of the same turn.
      */
     public void addTurnInfo(int fromEdge, int viaNode, int toEdge, long turnFlags) {
         // no need to store turn information
-        if (turnFlags == EMPTY_FLAGS)
+        if (turnFlags == EMPTY_FLAGS) {
             return;
+        }
+        BiFunction<Long, Long, Long> overwriteFunc = (oldFlags, newFlags) -> {
+            return newFlags;
+        };
+        setAndMergeTurnInfo(fromEdge, viaNode, toEdge, turnFlags, overwriteFunc);
+    }
 
+    /**
+     * Add a new turn cost entry.
+     * @param fromEdge edge ID
+     * @param viaNode node ID
+     * @param toEdge edge ID
+     * @param turnFlags flags to be written
+     * @param mergeFunction function which merges the old and the new flags as you prefer.
+     * This function takes two arguments – the old flags are the first, the new flags the
+     * second argument.
+     *
+     * See EdgeBasedRoutingAlgorithmTest for examples.
+     */
+    public void setAndMergeTurnInfo(int fromEdge, int viaNode, int toEdge, long turnFlags,
+            BiFunction<Long, Long, Long> mergeFunction) {
```

Comment:
Maybe instead of passing a function here we could convert this to an Enum that specifies the used mode like `MergeMode.OVERWRITE` and `MergeMode.OR`, that way we can add new modes in the future and still continue using JDK 7?